src/SmartHotel.Clients/SmartHotel.Clients/Services/IoT/RoomDevicesDataService.cs

```diff
@@ -1,32 +1,38 @@
 ﻿using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
+using SmartHotel.Clients.Core.Extensions;
 using SmartHotel.Clients.Core.Helpers;
 using SmartHotel.Clients.Core.Models;
+using SmartHotel.Clients.Core.Services.Request;
 
 namespace SmartHotel.Clients.Core.Services.IoT
 {
 	public class RoomDevicesDataService : IRoomDevicesDataService
 	{
-		// TODO: Probably best to get this in AppSettings
+	    // TODO: Probably best to get this in AppSettings
 		private readonly TimeSpan _sensorDataPollingInterval = TimeSpan.FromSeconds( 5 );
 		private readonly string _roomDevicesApiEndpoint;
 		private readonly string _roomId;
 		private Timer _sensorDataPollingTimer;
 
-		private readonly ConcurrentDictionary<string, DeviceSensorData>
+	    private readonly IRequestService _requestService; 
+
+        private readonly ConcurrentDictionary<string, DeviceSensorData>
 			_currentSensorDataBySensorDataType = new ConcurrentDictionary<string, DeviceSensorData>();
 
-		public RoomDevicesDataService()
+		public RoomDevicesDataService(IRequestService requestService)
```

Comment:
The BookingHotelViewModel shows how to get the auth token to use in api calls.

Have an IAuthenticationService get injected into the ctor